File: clojure/bob/bob.clj
(ns bob (:require [clojure.string :refer [upper-case blank?]]))


(defn is-yell?
	[string]
	(and (if (re-seq #"[A-Z]" string) true false) (= (upper-case string) string)))

(defn is-question?
	[string]
	(.endsWith string "?"))

(defn is-without?
	[string]
	(blank? string))

(defn response-for
	[string]
	(cond
		(is-without? string) "Fine. Be that way!"
		(is-yell? string) "Woah, chill out!"
		(is-question? string) "Sure."
		:else "Whatever."))

File: haskell/bob/Bob.hs
module Bob (responseFor) where

import Data.Char as C

responseFor :: String -> String
isQuestion :: String -> Bool
isYell :: String -> Bool
isWithout :: String -> Bool

responseFor str
	| isWithout str = "Fine. Be that way!"
	| isYell str = "Woah, chill out!"
	| isQuestion str = "Sure."
	| otherwise = "Whatever."

isQuestion str = last str == '?'

isYell str = any isAlpha str && all isUpper (filter isAlpha str)

isWithout str = all isSpace str
